cluster/src/main/scala/kanaloa/reactive/dispatcher/ClusterAwareBackend.scala

```diff
@@ -0,0 +1,90 @@
+package kanaloa.reactive.dispatcher
+
+import java.util.UUID
+
+import akka.actor._
+import akka.cluster.routing.{ClusterRouterGroup, ClusterRouterGroupSettings}
+import akka.routing._
+import akka.pattern.ask
+import akka.util.Timeout
+import kanaloa.reactive.dispatcher.ClusterAwareBackend.RouteeRef
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class ClusterAwareBackend(
+  actorRefPath: String,
+  role:         String,
+  routingLogic: RoutingLogic = RoundRobinRoutingLogic()
+)(implicit
+  system: ActorSystem,
+  timeout: Timeout = 1.seconds) extends Backend {
+
+  private[dispatcher] lazy val router: ActorRef = {
+    val routerProps: Props = ClusterRouterGroup(
+      RoundRobinGroup(List("/user/" + actorRefPath)),
+      ClusterRouterGroupSettings(
+        totalInstances = 100,
+        routeesPaths = List("/user/" + actorRefPath),
+        allowLocalRoutees = false, useRole = Some(role)
+      )
+    ).props()
+    system.actorOf(routerProps, s"clusterAwareBackendInternalRouter-$actorRefPath-$role-" + UUID.randomUUID().toString)
```

Comment:
Do you think we really should tack on a UUID, they can get long, and this is an already decently long name.  I understand why you need this random bit at the end, as to avoid name clashes, but do we need something this large?